### PR TITLE
UI: show segment debug details when segment is in bad state

### PR DESCRIPTION
**Pull request by jayeshchoudhary**
### What does this PR do?
- show segment debug details when segment is in bad state

### Issue
- One of the recurring issue is that if a segments show as BAD status, users have no idea why.  
- To get more info about error state then we need to check the debug API or check the server log for the segment.
- We need to have some way in UI so that users can get immediate feedback on why the segment is in BAD state

### Description
- Show a help icon besides `Bad` state. This also helps grab user attention when Bad state is visible
- On click of help icon a dialog will open with segment level debug details.

### Screenshots

https://user-images.githubusercontent.com/41536903/199269107-7de1f9cb-d5e7-43f7-bfc9-f9b13613444f.mp4

### CODE REVIEW

**Comment by jayeshchoudhary** [comment]
cc - @npawar @mayankshriv @saurabhd336 @joshigaurava

**Comment by jayeshchoudhary** [comment]
#9699 should be merged before this PR

**Review by jadami10 — COMMENTED** [review]
equally awesome as the previous PR. left some early feedback/question

File: pinot-controller/src/main/resources/app/utils/axios-config.ts [hunk 1]
```diff
@@ -65,4 +65,11 @@ baseApi.interceptors.response.use(handleResponse, handleError);
...
 export const transformApi = axios.create({baseURL: '/', transformResponse: [data => data]});
 transformApi.interceptors.request.use(handleConfig, handleError);
-transformApi.interceptors.response.use(handleResponse, handleError);
\ No newline at end of file
+transformApi.interceptors.response.use(handleResponse, handleError);
+
+// baseApi axios instance does not throw an error when API fails hence the control will never go to catch block
+// changing the handleError method of baseApi will cause current UI to break (as UI might have not handle error properly)
+// creating a new axios instance baseApiV1 which can be used when adding new API's
+export const baseApiV1 = axios.create({ baseURL: '/' });
```
[the remark this answers is not in the document]
Comment (jayeshchoudhary) [reply pinot-controller/src/main/resources/app/utils/axios-config.ts:73]: ack
Yes, it is not used in this PR but I faced this situation a lot of times where I wanted to notify the user that the API failed hence no data shown.

File: pinot-controller/src/main/resources/app/components/SegmentStatusRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            open={errorDetailsVisible}
+            handleClose={handleHideErrorDetails}
+            showOkBtn={false}
+          >
+            {/* Loading */}
+            {!segmentDebugDetails && (
+              <Box display="flex" justifyContent="center"><CircularProgress /></Box>
+            )}
+            {segmentDebugDetails && (
+              <CustomCodemirror
```
[the remark this answers is not in the document]
Comment (jayeshchoudhary) [reply pinot-controller/src/main/resources/app/components/SegmentStatusRenderer.tsx:177]: nice catch. updated 👍

**Comment by joshigaurava** [comment]
Looks good (once existing comments are addressed and https://github.com/apache/pinot/pull/9699 is merged).

**Review by jadami10 — COMMENTED** [review]
i'll review this again once the parent PR is merged and this is rebased so the changes are clearer

**Review by npawar — APPROVED** [review]

**Comment by Jackie-Jiang** [comment]
This is great!
When I try it out, I find a small issue (not sure if it is already fixed). When there are multiple pages of segments, and when I go to the second page, the color of the segment states are the same as the first page, and not match the actual segment state. @jayeshchoudhary

**Comment by jayeshchoudhary** [comment]
> This is great! When I try it out, I find a small issue (not sure if it is already fixed). When there are multiple pages of segments, and when I go to the second page, the color of the segment states are the same as the first page, and not match the actual segment state. @jayeshchoudhary

Nice catch @Jackie-Jiang this is a UI bug, I missed this case.
I have fixed this issue in the PR #9891.

**Pull request merged**